src/CodeGen_LLVM.cpp

```diff
@@ -603,6 +608,12 @@ void CodeGen_LLVM::begin_func(LoweredFunc::LinkageType linkage, const std::strin
     BasicBlock *block = BasicBlock::Create(*context, "entry", function);
     builder->SetInsertPoint(block);
 
+    // Put the global buffers in the symbol table.
+    for (const auto &buf : input_module->buffers()) {
+        llvm::Value *buf_ptr = sym_get(buf.name() + ".buffer");
```

Comment:
I'll have to look at this a bit more to understand the ramifications. I think "foo" should be the allocation and "foo.buffer" should be the buffer, but perhaps I misunderstand the comment. This might also fix the bug I'm seeing on the buildbot with Hexagon JIT and the simulator.

We should ensure the allocations for global buffers are maximally aligned.